src/components/TextInputs/PinCodeTextInput.tsx

```diff
@@ -0,0 +1,76 @@
+import React, { Component } from 'react'
+import { StyleSheet, TextInputProps, View } from 'react-native'
+import NumberInput from './NumberInput'
+
+interface Props extends TextInputProps {
+  onChangeInput: (input: string) => void
+  authCode?: string
+}
+
+interface State {
+  digits: string[]
+  focused: boolean
+}
+
+export default class PinCodeTextInput extends Component<Props, State> {
+  private inputRefs: NumberInput[]
+
+  constructor(props: Props, state: State) {
+    super(props, state)
+    this.inputRefs = []
+    this.state = {
+      digits: ['', '', '', '', '', ''],
```

Comment:
Maybe we could have a constant with value `6` that is the length of the pin code. This constant would be declared outside of the class and used to create this initial array and to be iterated when needed.

```suggestion
      digits: [...Array(numberOfDigits)].map(_ => ''),
```

That way in the future if we want to change this `6` to another value of if we want to make it a prop, code would be easier to refactor.